packages/amplify_storage_s3/lib/method_channel_storage_s3.dart

```diff
@@ -35,24 +37,25 @@ class AmplifyStorageS3MethodChannel extends AmplifyStorageS3 {
     } on PlatformException catch (e) {
       if (e.code == "AmplifyAlreadyConfiguredException") {
         throw AmplifyAlreadyConfiguredException(
-          AmplifyExceptionMessages.alreadyConfiguredDefaultMessage,
-          recoverySuggestion: AmplifyExceptionMessages.alreadyConfiguredDefaultSuggestion);
+            AmplifyExceptionMessages.alreadyConfiguredDefaultMessage,
+            recoverySuggestion:
+                AmplifyExceptionMessages.alreadyConfiguredDefaultSuggestion);
       } else {
-        throw AmplifyException.fromMap(
-            Map<String, String>.from(e.details));
+        throw AmplifyException.fromMap(Map<String, String>.from(e.details));
       }
     }
   }
 
   @override
   Future<UploadFileResult> uploadFile(
-      {@required UploadFileRequest request}) async {
+      {required UploadFileRequest request}) async {
     try {
-      final Map<String, dynamic> data =
-          await _channel.invokeMapMethod<String, dynamic>(
+      final Map<String, dynamic>? data =
+          (await _channel.invokeMapMethod<String, dynamic>(
         'uploadFile',
         request.serializeAsMap(),
-      );
+      ));
+      if (data == null) throw nullMapFromMethodChannelException;
```

Comment:
Could you please explain the need for `nullMapFromMethodChannelException`?

Reply:
The NS upgrade treats any information received from MethodChannels as nullable.  

In our method_channel code we can either force cast this result, or throw an exception if the result is null.  It's better to throw an exception instead of force cast, thus we throw "nullMapFromMethodChannelException" in case the returned result is actually null.